pkg/core/blockchain.go

```diff
@@ -80,13 +80,21 @@ type Blockchain struct {
 
 	// cache for block verification keys.
 	keyCache map[util.Uint160]map[string]*keys.PublicKey
+
+	log *zap.Logger
 }
 
 type headersOpFunc func(headerList *HeaderHashList)
 
 // NewBlockchain returns a new blockchain object the will use the
 // given Store as its underlying storage.
-func NewBlockchain(s storage.Store, cfg config.ProtocolConfiguration) (*Blockchain, error) {
+func NewBlockchain(s storage.Store, cfg config.ProtocolConfiguration, log *zap.Logger) (*Blockchain, error) {
+	if log == nil {
+		log = zap.NewNop()
+	}
+
+	icLog = log
```

Comment:
What's this?

Reply:
Now I do see that it's defined for interops, but I don't think it's a good idea, better pass it into the `newInteropContext()`.

Reply:
Or take it from bc.log.